app/src/main/java/io/homeassistant/companion/android/settings/qs/ManageTilesFragment.kt

```diff
@@ -0,0 +1,226 @@
+    }
+
+    override fun onPrepareOptionsMenu(menu: Menu) {
+        super.onPrepareOptionsMenu(menu)
+
+        menu.findItem(R.id.get_help)?.let {
+            it.isVisible = true
+            it.intent = Intent(Intent.ACTION_VIEW, Uri.parse("https://companion.home-assistant.io/docs/integrations/android-quick-settings"))
+        }
+    }
+
+    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
+        setPreferencesFromResource(R.xml.quick_setting_tiles, rootKey)
+    }
+
+    override fun onResume() {
+        super.onResume()
+
+        val loader = IconPackLoader(requireContext())
+        iconPack = createMaterialDesignIconPack(loader)
+        iconPack.loadDrawables(loader.drawableLoader)
+        val settings = IconDialogSettings {
+            searchVisibility = IconDialog.SearchVisibility.ALWAYS
+        }
+        val iconDialog = IconDialog.newInstance(settings)
+
+        DaggerSettingsComponent.builder()
+            .appComponent((activity?.applicationContext as GraphComponentAccessor).appComponent)
+            .build()
+            .inject(this)
+
+        activity?.title = getString(R.string.tiles)
+        val tileDao = AppDatabase.getInstance(requireContext()).tileDao()
+        var tileList = tileDao.getAll()
+        var tileLabel = findPreference<EditTextPreference>("tile_label")?.text
+        var tileSubtitle = findPreference<EditTextPreference>("tile_subtitle")?.text
+        val tileEntityPref = findPreference<ListPreference>("tile_entity")
+        var tileId = findPreference<ListPreference>("tile_list")?.value
+        val resumeIcon = tileDao.get(tileId!!)?.iconId
+        val tileSavePref = findPreference<Preference>("tile_save")
+        var tileEntity = tileEntityPref?.value
+        var entityList = listOf<String>()
+
+        if (resumeIcon != null) {
+            val iconDrawable = iconPack.getIcon(resumeIcon)?.drawable
+            if (iconDrawable != null) {
+                val icon = DrawableCompat.wrap(iconDrawable)
+                icon.setColorFilter(resources.getColor(R.color.colorAccent), PorterDuff.Mode.SRC_IN)
+                findPreference<Preference>("tile_icon")?.let {
+                    it.icon = icon
+                    it.summary = resumeIcon.toString()
+                }
+            }
+        }
+
+        runBlocking {
+            try {
+                integrationUseCase.getEntities().forEach {
+                    val split = it.entityId.split(".")
+                    if (split[0].startsWith("script") || split[0].startsWith("scene"))
+                        entityList = entityList + it.entityId
+                }
+            } catch (e: Exception) {
+                Log.e(TAG, "Unable to fetch list of entity IDs", e)
+            }
+        }
+
+        if (entityList.isNotEmpty()) {
+            tileEntityPref?.entries = entityList.sorted().toTypedArray()
+            tileEntityPref?.entryValues = entityList.sorted().toTypedArray()
```

Comment:
This is sorting and creating the same array 2 times.

Reply:
ah good catch